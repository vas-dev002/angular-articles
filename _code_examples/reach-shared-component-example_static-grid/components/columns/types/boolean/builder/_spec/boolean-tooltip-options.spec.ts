import { ITooltipParams } from "@ag-grid-community/core";
import { TestBed } from "@angular/core/testing";
import { TranslateService, TranslateModule } from "@ngx-translate/core";

import { BooleanTooltipOptions } from "..";

declare const expect: jest.Expect;

describe("[Builder option]: BooleanTooltipOptions", () => {
	let booleanTooltipOptions: BooleanTooltipOptions;
	let translateService: TranslateService;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [TranslateModule.forRoot()],
			providers: [TranslateService]
		});

		translateService = TestBed.inject(TranslateService);
		booleanTooltipOptions = new BooleanTooltipOptions(translateService);
	});

	describe("[BooleanTooltipOptions instantiate]", () => {
		it("should be successfully created", () => {
			expect(booleanTooltipOptions).toBeTruthy();
			expect(booleanTooltipOptions).toMatchSnapshot();
		});
	});

	describe("[Method]: getterFunction", () => {
		it("should return the currect value if data is set", () => {
			const params = { value: "False" };

			const tooltipValueGetter = booleanTooltipOptions.tooltipValueGetter(params as unknown as ITooltipParams);
			expect(tooltipValueGetter).toEqual("False");
		});

		it("should return null", () => {
			const params = { value: null };

			const tooltipValueGetter = booleanTooltipOptions.tooltipValueGetter(params as unknown as ITooltipParams);
			expect(tooltipValueGetter).toBeNull();
		});
	});
});
