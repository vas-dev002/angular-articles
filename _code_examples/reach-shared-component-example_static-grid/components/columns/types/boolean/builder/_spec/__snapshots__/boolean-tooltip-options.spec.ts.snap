// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Builder option]: BooleanTooltipOptions [BooleanTooltipOptions instantiate] should be successfully created 1`] = `
BooleanTooltipOptions {
  "langService": TranslateService {
    "_langs": Array [],
    "_onDefaultLangChange": EventEmitter_ {
      "__isAsync": false,
      "_isScalar": false,
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "_onLangChange": EventEmitter_ {
      "__isAsync": false,
      "_isScalar": false,
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "_onTranslationChange": EventEmitter_ {
      "__isAsync": false,
      "_isScalar": false,
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "_translationRequests": Object {},
    "_translations": Object {},
    "compiler": TranslateFakeCompiler {},
    "currentLoader": TranslateFakeLoader {},
    "extend": false,
    "isolate": false,
    "missingTranslationHandler": FakeMissingTranslationHandler {},
    "parser": TranslateDefaultParser {
      "templateMatcher": /\\{\\{\\\\s\\?\\(\\[\\^\\{\\}\\\\s\\]\\*\\)\\\\s\\?\\}\\}/g,
    },
    "pending": false,
    "store": TranslateStore {
      "currentLang": undefined,
      "langs": Array [],
      "onDefaultLangChange": EventEmitter_ {
        "__isAsync": false,
        "_isScalar": false,
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
      "onLangChange": EventEmitter_ {
        "__isAsync": false,
        "_isScalar": false,
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
      "onTranslationChange": EventEmitter_ {
        "__isAsync": false,
        "_isScalar": false,
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
      "translations": Object {},
    },
    "useDefaultLang": true,
  },
  "tooltipComponentFramework": [Function],
  "tooltipValueGetter": [Function],
}
`;
